src/realm/parser/query_parser.hpp

```diff
@@ -151,6 +190,11 @@ class ArgumentConverter : public Arguments {
         return m_arguments[index];
     }
 
+    DataType type_for_argument(size_t i) override
+    {
+        return m_ctx.get_type_of(at(i));
```

Comment:
This works for our internal AnyContext, but will require the realm-js [NativeAccessor](https://github.com/realm/realm-js/blob/e67a78bb5ad32245a2b531d73ed80baa9570e2a4/src/js_object_accessor.hpp#L46) to implement `get_type_of()`. (the call site is [here](https://github.com/realm/realm-js/blob/master/src/js_results.hpp#L168-L172)) That seems ok to me, and it seems to me that they should be able to use Mixed to simplify a bunch of their code. But if we are going to require changes at the SDK level, what do you think about requiring them to implement the MixedArguments arguments interface? If we can expect MixedArguments everywhere, that would simplify all the hoops we have to jump through here at the core level.